### PR TITLE
Refactor metrics TimeWindow 

### PR DESCRIPTION
**Pull request by mytang0**
## What is the purpose of the change
issue #11637 
1. Abstract sliding window
2. Refactor TimeWindow Counter and Quantile

## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GitHub issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [x] Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Check if is necessary to patch to Dubbo 3 if you are work on Dubbo 2.7
- [x] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Add some description to [dubbo-website](https://github.com/apache/dubbo-website) project if you are requesting to add a feature.
- [x] GitHub Actions works fine on your own branch.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

**Comment by mytang0** [comment]
PTAL @songxiaosheng

**Comment by songxiaosheng** [comment]
> PTAL @songxiaosheng  @Henrik-Yao 

please  wait,thanks

**Comment by mytang0** [comment]
Please take a review @AlbumenJ

File: dubbo-common/src/main/java/org/apache/dubbo/common/utils/TimeUtils.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+                } catch (Exception ignored) {
+                    //
+                }
+            }
+        });
+        ticker.setDaemon(true);
+        ticker.setName("time-millis-ticker-thread");
+        ticker.start();
+        Runtime.getRuntime().addShutdownHook(new Thread(ticker::interrupt));
+    }
```
Comment (AlbumenJ) [inline dubbo-common/src/main/java/org/apache/dubbo/common/utils/TimeUtils.java:50]: 1. Prevent add complex logic in the process of class initialization
2. Start on demand

**Review by AlbumenJ — CHANGES_REQUESTED** [review]

File: dubbo-common/src/main/java/org/apache/dubbo/common/utils/TimeUtils.java (hunk 1 above)
Comment (mytang0) [reply dubbo-common/src/main/java/org/apache/dubbo/common/utils/TimeUtils.java]: > 1. Prevent add complex logic in the process of class initialization
> 2. Start on demand

Fixed.

File: dubbo-common/src/main/java/org/apache/dubbo/common/utils/TimeUtils.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    private static volatile boolean isTickerAlive = false;
+
+    private TimeUtils() {
+    }
+
+    public static long currentTimeMillis() {
+        if (!isTickerAlive) {
+            startTicker();
+        }
+        return currentTimeMillis;
```
Comment (AlbumenJ) [inline dubbo-common/src/main/java/org/apache/dubbo/common/utils/TimeUtils.java:38]: Should fallback if `currentTimeMillis` is un-available here?

**Review by AlbumenJ — CHANGES_REQUESTED** [review]

**Review by songxiaosheng — APPROVED** [review]
LGTM

File: dubbo-common/src/main/java/org/apache/dubbo/common/utils/TimeUtils.java (hunk 2 above)
Comment (mytang0) [reply dubbo-common/src/main/java/org/apache/dubbo/common/utils/TimeUtils.java:49]: > Should fallback if `currentTimeMillis` is un-available here?

Good. Submitted.

**Review by AlbumenJ — APPROVED** [review]

**Pull request merged**